pkg/ccl/kvccl/kvfollowerreadsccl/followerreads.go

```diff
@@ -203,6 +205,49 @@ func (o *followerReadOracle) useClosestOracle(
 // whether a query can be used with a follower read.
 var followerReadOraclePolicy = replicaoracle.RegisterPolicy(newFollowerReadOracle)
 
+type bulkOracle struct {
+	cfg       replicaoracle.Config
+	locFilter roachpb.Locality
+}
+
+var _ replicaoracle.Oracle = bulkOracle{}
+
+// NewBulkOracle returns an oracle for planning bulk operations, which will plan
+// balancing random across all replicas if follower reads are enabled.
+func NewBulkOracle(cfg replicaoracle.Config, locFilter roachpb.Locality) replicaoracle.Oracle {
+	return bulkOracle{cfg: cfg, locFilter: locFilter}
+}
+
+// ChoosePreferredReplica implements the replicaoracle.Oracle interface.
+func (r bulkOracle) ChoosePreferredReplica(
+	ctx context.Context,
+	_ *kv.Txn,
+	desc *roachpb.RangeDescriptor,
+	leaseholder *roachpb.ReplicaDescriptor,
+	_ roachpb.RangeClosedTimestampPolicy,
+	_ replicaoracle.QueryState,
+) (_ roachpb.ReplicaDescriptor, ignoreMisplannedRanges bool, _ error) {
+	if leaseholder != nil && !checkFollowerReadsEnabled(uuid.UUID{} /*not used*/, r.cfg.Settings) {
+		return *leaseholder, false, nil
```

Comment:
if leaseholder is nil and follower reads disabled, we will panic.
I don't think that's what we want.